main/main.c

```diff
@@ -312,6 +312,7 @@ static void php_binary_init(void)
 			binary_location = NULL;
 		}
 	} else {
+		free(binary_location);
```

Comment:
I think this is freeing an uninitialized pointer. Isn't the allocation of binary_location only in the if branch?

Reply:
The Travis segfault on sapi/fpm/tests/001.phpt *might* be related to this.

Reply:
You're right! I think I was intending to hoist the allocation of `binary_location` out of the conditional preprocessor blocks, but initialising `binary_location` to `NULL` is probably better here because you can remove that `else` branch entirely (and accidentally delete a necessary brace... ooops!)

Reply:
Ugh, that's not going to work, either... I give up. I'm not really suitable for this team, anyway.